Support timeline background press for creating new events

ethanshar opened the pull request:
- Support timeline background (long) press for creating new events - currently this is a very simple behavior till we have the final UX
- Enable TS in the Timeline example screen 
- Added usage example
- Move logic to a presenter file and add tests

lidord-wix reviewed (commented):
Looks good overall, I left a few comments, and here are more general comments:  
- IMO, the background long press should "floor" the time of the created event, for example - pressing on 13:50 should set a new event at 13:30-14:30 and not at 14:00-15:00.
- It doesn't reproduce all the time, but sometimes when I long-press the background it doesn't create a new event 
- There's a small conflict with master to resolve

ethanshar commented on src/timeline/Timeline.tsx:
```diff
@@ -6,22 +6,45 @@ import map from 'lodash/map';
+  /**
+   * Pass to handle creation of a new event by long press on the timeline background
+   */
+  onEventCreate?: TimelineHoursProps['onBackgroundLongPress'];
```
I will actually change it a little anyway now that I have the UX guidelines. 
I will have two callbacks 
onLongPress -> this should generate a new event placeholder
onPressOut -> this should invoke the event creation callback 
This is how it works in Calendars apps, so I'll try to mimic that

ethanshar replied:
Fixed, you can see how I use it now in the example screen

ethanshar commented on example/src/screens/timelineCalendar.tsx:
```diff
@@ -101,7 +100,24 @@ export default class TimelineCalendarScreen extends Component {
+      start: `${currentDate} ${timeString}`,
+      end: `${currentDate} ${(timeObject.hour + 1).toString().padStart(2, '0')}:${timeObject.minutes
+        .toString()
+        .padStart(2, '0')}:00`,
```
Fixed

lidord-wix replied:
Sorry, I wasn't clear enough.. I meant to add consts like: 
```
const hourString  = (timeObject.hour + 1).toString().padStart(2, '0');
const minutesString = timeObject.minutes.toString().padStart(2, '0');
```

ethanshar replied:
Cool. Fixed 👍

ethanshar commented:
> Looks good overall, I left a few comments, and here are more general comments:
> 
> * IMO, the background long press should "floor" the time of the created event, for example - pressing on 13:50 should set a new event at 13:30-14:30 and not at 14:00-15:00.
> * It doesn't reproduce all the time, but sometimes when I long-press the background it doesn't create a new event
> * There's a small conflict with master to resolve

- Yes, I agree. I fixed the rounding logic
- It works fine for me, see now with the new creating logic if you still have issues
- Fixed ✔️

ethanshar commented on src/timeline/Timeline.tsx:
```diff
@@ -34,7 +34,11 @@ export interface TimelineProps {
+  /**
+   * Pass to handle creation of a new event by long press out on the timeline background
+   */
+   onBackgroundLongPressOut?: TimelineHoursProps['onBackgroundLongPressOut'];
```
Fixed (:
